src/dotdirdeps.cpp

```diff
@@ -136,8 +136,7 @@ void writeDotDirDepGraph(FTextStream &t,const DirDef *dd,bool linkRelations)
         //    shortName().data(),
         //    !usedDir->isParentOf(this)
         //    );
-        if (dir!=usedDir && dir->parent()==usedDir->parent() &&
-            !usedDir->isParentOf(dd))
```

Comment:
might this have something to do with recursive directories (maybe based on symbolic links)?

Reply:
Do you mean, that a chain of `parent()` may be cyclic because of (symbolic) links in the file system?

Reply:
Yes this is what I meant, it is a wild guess though.

Reply:
I did some speculative tests in order to break it. But with no success: Apparently, the symbolic links are not used. Although I had them not excluded. This is my Doxyfile:

```
EXTRACT_ALL            = YES
RECURSIVE              = YES
HAVE_DOT               = YES
DOT_CLEANUP            = NO
EXCLUDE_SYMLINKS       = NO
DOT_IMAGE_FORMAT       = PNG
```

```
$ tree src/
src/
└── usedDir
    ├── dependee.h
    └── parent
        ├── dir
        │   └── dependent.h
        ├── link -> ../../usedDir/
        ├── otherDir
        │   └── dependee3_original.h
        ├── otherDir2
        │   └── dependee3.h -> otherDir/dependee3_original.h
        └── usedDir2
            ├── dependee2.h
            └── dependee.h -> usedDir/dependee.h
```

`dependent.h`:
```
#include "dependee.h"
#include "dependee2.h"
#include "dependee3.h"
```

The result is, that `dependee3.h`, `usedDir2/dependee.h`, `link` do not appear in the File List page neither in the directory dependency graphs.

[recursive_link_example.zip](https://github.com/doxygen/doxygen/files/5766626/recursive_link_example.zip)

Reply:
I recently wondered as well why this check was there. I also couldn't think of any reason.